packages/cozy-client/src/CozyClient.js

```diff
@@ -467,6 +467,19 @@ class CozyClient {
     // storing the document. This makes the data easier to manipulate for the front-end.
     const relationshipsByDocId = {}
 
+    for (const def of documents) {
+      const docIdAndRels = queryDefToDocIdAndRel.get(def)
+      for (const docIdAndRel of docIdAndRels) {
+        if (docIdAndRel) {
+          const [docId, relName] = docIdAndRel
+          relationshipsByDocId[docId] = relationshipsByDocId[docId] || {}
+          relationshipsByDocId[docId][relName] = responseToRelationship({
+            data: def
+          })
+        }
+      }
+    }
+
```

Comment:
Mmh I think this for and the following could be extracted into a method `reconstructRelationships(queryDefToDocIdAndRel, definitions, documents, responses)`